### PR TITLE
feat: add or filtering support

### PR DESCRIPTION
**Pull request by massimo-ua**
Added support of OR filter conditions
This adds support of combining multiple filters with "or" condition. Conditions can be passed as array of filter fragments
Any feedback are appreciated

### CODE REVIEW

**Comment by alechirsch** [comment]
@massimo-ua Briefly looking at this, I like the concept and API. I will be able to give it a further look next week. Thanks for your work on this.

How do you propose that an API user will use "or" filtering through a query parameter?

**Comment by massimo-ua** [comment]
> @massimo-ua Briefly looking at this, I like the concept and API. I will be able to give it a further look next week. Thanks for your work on this.
> 
> How do you propose that an API user will use "or" filtering through a query parameter?

@alechirsch Actually I have working implementation forked from kideh88/node-jsonapi-query-parser
and query is looks like this:
```
?filter[or][0][like][mirrors.name]=test3
&filter[or][0][type]=static
&filter[or][1][fid]=1041
```
where number that follows [or] is used as index to group properties in one filter condition
```
filter: {
    or: [
        { like: { mirrors.name: test3 }, type: static },
        { fid: 1041 }
    ]
}
```

File: test/postgres.js [hunk 1]
```diff
@@ -552,6 +552,54 @@ describe('bookshelf-jsonapi-params with postgresql', () => {
                     done();
                 });
         });
+
+    });
+
+    describe('passing in "or" filtering', () => {
+        it('should return results for "or" filters combination', (done) => {
```
Comment (alechirsch) [inline test/postgres.js:580]: Can you add a few tests for combing base filters with or filters?

Comment (alechirsch) [reply test/postgres.js:580]: And some json filtering inside of `or`

**Comment by alechirsch** [comment]
@massimo-ua There are merge conflicts that need to be resolved

File: test/postgres.js (hunk 1 above)
Comment (massimo-ua) [reply test/postgres.js:580]: @alechirsch Added few test for filter combinations. Just thought that actually every filter it's a combination of "or" filters with one ("base filter") or more ("or filter") filter fragments. So every filter can be described as array of filter fragments

**Comment by massimo-ua** [comment]
> @massimo-ua There are merge conflicts that need to be resolved
Yep. I'm going to manage this in a few days. Do you have any recommendation?

**Comment by alechirsch** [comment]
@massimo-ua I added support for allowing the user to provide arrays of strings for values instead of only comma separated strings. So your assumption for if a value is an array then it is for `or` type does not stand. You need to explicitly check for the `or` key instead of checking if it is an array.

File: src/index.js [hunk 2]
```diff
@@ -98,7 +99,47 @@ export default (Bookshelf, options = {}) => {
...
+                    internals.buildDependenciesHelper(typeKey, relationHash);
+                });
+            }
+        };
+        /**
+         * Build a query for array of relational dependencies of filtering
+         * @param   filterList {array}
+         * @param   relationHash {object}
+         */
+        internals.buildArrayLikeFilterValue = (filterList, relationHash) => {
```
Comment (alechirsch) [inline src/index.js]: Can we rename this to be more specific that it is only for `or` filters?

Comment (massimo-ua) [reply src/index.js]: Done

**Comment by alechirsch** [comment]
@massimo-ua Released as 1.5.0

**Pull request merged**